bytebufferpool/bytebuffer_test.go

```diff
@@ -155,3 +155,111 @@ func TestByteBufferCopy(t *testing.T) {
 		t.Fatal("Copy content is wrong")
 	}
 }
+
+func TestCopyWithIdleDuration(t *testing.T) {
+
+	b := Get()
+	b.Reset()
+	src := GetSleepByteBuffer()
+	dst := Get()
+	dst.Reset()
+
+	s := "hello world!"
+	src.Write([]byte(s))
+	src.SetTime(0)
+
+	n, err := b.CopyWithIdleDuration(dst, src, 0)
+	if err != nil {
+		t.Fatalf("unexpected err: %s", err.Error())
+	}
+	if int(n) != src.Len() {
+		t.Fatal("Copy size is wrong")
+	}
+	if !bytes.Equal(dst.Bytes(), src.Bytes()) {
+		t.Fatal("Copy content is wrong")
+	}
+
+	dst.Reset()
+	src.Reset()
+	src.SetTime(1 * time.Second)
+	n, err = b.CopyWithIdleDuration(dst, src, 2*time.Second)
+	if err != nil {
+		t.Fatalf("unexpected err: %s", err.Error())
+	}
+	if int(n) != src.Len() {
+		t.Fatal("Copy size is wrong")
+	}
+	if !bytes.Equal(dst.Bytes(), src.Bytes()) {
+		t.Fatal("Copy content is wrong")
+	}
+
+	dst.Reset()
+	src.Reset()
+	src.SetTime(4 * time.Second)
+	n, err = b.CopyWithIdleDuration(dst, src, 1*time.Second)
+	if err == nil {
+		t.Fatal("expected err: idle time out")
+	}
+	if !strings.Contains(err.Error(), "idle time out") {
+		t.Fatal("expected err: idle time out, but get unexpected error")
```

Comment:
displacy the unexpected error for tester:

`t.Fatal("expected err: idle time out, but get unexpected error: ", err)`